src/types/ScreenInfoUiaProvider.h

```diff
@@ -77,15 +82,18 @@ namespace Microsoft::Console::Types
         IFACEMETHODIMP get_DocumentRange(_COM_Outptr_result_maybenull_ ITextRangeProvider** ppRetVal);
         IFACEMETHODIMP get_SupportedTextSelection(_Out_ SupportedTextSelection* pRetVal);
 
-        HWND GetWindowHandle() const;
+        std::optional<HWND> GetWindowHandle() const;
         void ChangeViewport(const SMALL_RECT NewWindow);
 
     private:
         // Ref counter for COM object
         ULONG _cRefs;
 
         // weak reference to uia parent
-        WindowUiaProviderBase* const _pUiaParent;
+        std::optional<WindowUiaProviderBase*> const _pUiaParent;

```

Comment:
Now you have to check two things. You have to check if the optional.has_value() and then when you get optional.value(), you have to check that for null too. Unless there's some template magic tying the optional's has_value call to the filled value of nullptr for a pointer type.

Given a pointer is already implicitly nullable, you should just check that, IMO.